File: chap03_ionchannels/ED_profiles.py

#   -------------------------------------------------------------------
# 
#    Plot profiles from the PNP equations.
# 
#    For Chapter 3, Section 3.3.1 of
#    Keener and Sneyd, Mathematical Physiology, 3rd Edition, Springer.
#  
#    Written by James Keener and James Sneyd
#  
#   ------------------------------------------------------------------- 

import numpy as np
import matplotlib.pyplot as plt

# Set plot parameters
plt.rcParams.update({
    'axes.titlesize': 20,
    'axes.linewidth': 2.0,
    'lines.linewidth': 2.0
})

plt.rcParams['text.usetex'] = True

# Constants
y = np.linspace(0, 1, 100)
ui = 50 / 550
ue = 500 / 550
v = 1
J1 = v * (ui - ue * np.exp(-v)) / (1 - np.exp(-v))
K1 = ui - J1 / v

# Profiles
cprofile = J1 / v + K1 * np.exp(v * y)
vprofile = v - v * y
cprofilelong = ui + (ue - ui) * y
vprofilelong = -(v / np.log(ue / ui)) * np.log(ui / ue + (1 - ui / ue) * y)

# Plot cprofile and vprofile
plt.figure(1)
plt.plot(y, cprofile, label='$u_1$')
plt.plot(y, vprofile, label='$\psi$')
plt.xlabel('y')
plt.legend()

# Plot cprofilelong and vprofilelong
plt.figure(2)
plt.plot(y, cprofilelong, label='$u_1$')
plt.plot(y, vprofilelong, label='$\psi$')
plt.xlabel('y')
plt.legend()

File: chap03_ionchannels/ED_profiles.m
%  -------------------------------------------------------------------
%
%   Plot profiles from the PNP equations.
%
%   For Chapter 3, Section 3.3.1 of
%   Keener and Sneyd, Mathematical Physiology, 3rd Edition, Springer.
% 
%   Written by James Keener and James Sneyd
% 
%  ------------------------------------------------------------------- 

clear all
close all
clc
set(0,                           ...
   'defaultaxesfontsize', 20,   ...
   'defaultaxeslinewidth', 2.0, ...
   'defaultlinelinewidth', 2.0);


y = linspace(0,1,100);
ui = 50/550;
ue = 500/550;
v = 1;
J1 = v*(ui - ue*exp(-v))/(1-exp(-v));
K1 = ui - J1/v;

cprofile = J1/v + K1*exp(v*y);
vprofile= v-v*y;
profilex = y;
cprofilelong = ui +(ue-ui)*y;
vprofilelong = -(v/log(ue/ui))*log(ui/ue + (1-ui/ue)*y);

figure(1)
    plot(y,cprofile,y,vprofile)
    xlabel('y')
    legend('u_1','\psi')

figure(2)
    plot(y,cprofilelong,y,vprofilelong)
    xlabel('y')
    legend('u_1','\psi')